### PR TITLE
use outputPath to check if thi is a new file

### PR DESCRIPTION
**Pull request by ZeeD**
related to #690

### CODE REVIEW

File: src/Program.ts [hunk 1]
```diff
@@ -1388,13 +1388,13 @@ export class Program {
...
 
-        const processedFiles = new Set<File>();
+        const processedFiles = new Set<string>();
 
         const transpileFile = async (srcPath: string, outputPath?: string) => {
             //find the file in the program
             const file = this.getFile(srcPath);
             //mark this file as processed so we don't process it more than once
-            processedFiles.add(file);
+            processedFiles.add(outputPath);
```
Comment (TwitchBronBron) [inline src/Program.ts:1397]:
```suggestion
            processedFiles.add(outputPath.toLowerCase());
```

File: src/Program.ts [hunk 2]
```diff
@@ -1437,9 +1437,10 @@ export class Program {
             for (const key in this.files) {
                 const file = this.files[key];
                 //this is a new file
-                if (!processedFiles.has(file)) {
+                const outputPath = getOutputPath(file);
+                if (!processedFiles.has(outputPath)) {
```
Comment (TwitchBronBron) [inline src/Program.ts:1441]:
```suggestion
                if (!processedFiles.has(outputPath.toLowerCase())) {
```

**Review by TwitchBronBron — CHANGES_REQUESTED** [review]
Looks pretty good. Just need to `toLowerCase` those lookups and then it'll be good to go.

**Comment by ZeeD** [comment]
yeah, sorry I forgot to add the `.toLowerCase()` invocation

**Review by TwitchBronBron — APPROVED** [review]

**Pull request merged**